sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -697,6 +813,103 @@ public TableReference getTable() {
       ValueProvider<TableReference> provider = getTableProvider();
       return provider == null ? null : provider.get();
     }
+
+    public ReadGenericRecords<T> withCoder(Coder<T> coder) {
+      return toBuilder().setCoder(coder).build();
+    }
+
+    /**
```

Comment:
I'd recommend to avoid the duplication of javadocs and document all of these methods as "Like {@link Read#sameMethod}", to reduce the burden for future maintainers.

Reply:
👍